Update

developer9998 opened the pull request:
Recently decided to go back through some of the older versions of the mod, and looking through them I found tons of content that have either stayed the exact same since the release (Mainly through 2021) or have been replaced which content which many others consider as lower quality or worse in a way. With this I've went out my way to edit some of the content I mentioned prior which includes:
- Reverting the Computer model (More space, easier to work with aspect ratio, non-bulky, etc.)
- Changing much of the Game Settings tabs (Mic, Voice, etc.)
- More side changes

Also due to these changes I've bumped the version to 1.6.0, but you can change this and obviously any of this if you want.

## Computer
<img src="https://user-images.githubusercontent.com/81720436/232342661-254b1540-31e2-4fb7-91b9-e6b09015d048.gif" width=40% height=40%><br>
- Changed the model (As mentioned)
- Changed key press colours for custom keys (Back, Up, Down, Left, Right)
- Made interacting with the keyboard much easier

## Room tab
<img src="https://user-images.githubusercontent.com/81720436/232343113-5ece5501-5428-42eb-be84-823534118300.png" width=40% height=40%><br>
- Made text show for private rooms and group joins

## Name tab
<img src="https://user-images.githubusercontent.com/81720436/232343266-5d3080fb-d8fc-4eec-b62d-8c5f6c9115cf.png" width=40% height=40%><br>
- Made text show for when you get an error (Empty name, blank name, name too long, etc.)
- Removed spaces in output name when set (Seen in image)

## (Color/Colour tab is the same)
## Turn tab
<img src="https://user-images.githubusercontent.com/81720436/232343528-ea19421b-6b0a-42fe-9fdb-cf687c8497c7.png" width=40% height=40%><br>
- Reorganized page

## Mic tab
<img src="https://user-images.githubusercontent.com/81720436/232343554-fd8b7d8e-eb2a-4f83-bff8-84535f3d7fa3.png" width=40% height=40%><br>
- Reorganized page

## Group tab
<img src="https://user-images.githubusercontent.com/81720436/232343644-123b9a34-49df-4744-bb07-47880f4838de.png" width=40% height=40%><br>
- <b>Actually fixed group joining, pretty sure it's been broken for months</b>
- Reorganized page

## Voice tab
<img src="https://user-images.githubusercontent.com/81720436/232343696-19a028b2-1ac5-41f8-a5b8-e3edb82f3c71.png" width=40% height=40%><br>
- Reorganized page

## Items tab (WAS Visuals tab)
<img src="https://user-images.githubusercontent.com/81720436/232343716-b43f8502-ee8b-42ce-a79d-697ff2759a7c.png" width=40% height=40%><br>
- Reorganized page

## Credits tab
<img src="https://user-images.githubusercontent.com/81720436/232343745-f7c1492d-f76e-4572-875e-b2246f679216.png" width=40% height=40%><br>
- Added an actual page system
- Made page sync up with the base game computer

## NEW! Support tab
<img src="https://user-images.githubusercontent.com/81720436/232343844-0851ace3-2ae7-45c7-9d81-bb856212e761.png" width=40% height=40%><br>

## Other changes
- Made Details page also display "User bans yesterday"
- Made Commandline view display "Commandline" instead of "Monkeshell"

developer9998 committed:
Added some adjustments

-- Base Game Interface --
- Limited colour values when you want to get and set them to prevent impossible colours
- Simplified name checks
- Made networked player name text change once you set your name
- Optimized the InitializeNoobMaterial method

-- NameSettingView --
- Made name set when you press the enter key instead of the back key
- Removed empty name checks and moved them to BaseGameInterface

developer9998 committed:
Base view overhaul

Here we go..

BaseGameInterface
- Put sections under regions for better organization
- Made methods that used GorillaComputer.instance use the CheckForComputer method instead
- Updated the group join method

CustomComputer
- Optimized the process of getting the Cube mesh

DetailsView
- Updated the spacing
- Added users banned

ColorSettingView
- Updated the header
- Optimized key inputs

CreditsView
- Made it up to date with the credits used for the main computer
- Optimized key inputs

GroupView
- Updated the selection indicator
- Optimized key inputs

ItemSettingsView
- Updated the selection indicator
- Renamed display name to Item instead of Visual
- Optimized key inputs

JoinRoomView
- Optimized RoomCallbacks object
- Optimized key inputs

MicSettingsView
- Updated the selection indicator
- Cleaned up most the page
- Optimized key inputs

NameSettingView
- Added an indicator for when you change your name
- Optimized key inputs

QueueSettingsView
- Added note on top
- Optimized key inputs (For some reason)

TurnSettingView
- Updated the selection indicator
- Cleaned up most the page
- Optimized key inputs

VoiceSettingsView
- Updated the selection indicator
- Cleaned up most the page
- Optimized key inputs

GameSettingsView
- Moved the order of the elements to fit the base game computer
- Made repeat based on views

ModListView
- Put the MethodInfo detection in a try statement which does literally nothing

[Other Changes]
- Added SupportView

developer9998 committed:
Hotfix

PluginInfo
- Bumped version

JoinRoomView
- Fixed header

developer9998 committed:
Minor update

CommandRegistrar
- Added several checks for the SetName and Join commands

BaseGameInterface
- Added checks for joining rooms and setting names

DetailsView
- Fixed formatting
- Made spacing fit the entire page

CustomKeyboardKey
- Reduced much of the cooldown
- Optimized key press method

CustomComputer
- Added method for simulating a key press with a EKeyboardKey enum

developer9998 committed:
Update

developer9998 committed:
Updates + Bump to 1.6.0

This updates and/or fixes much of the older tabs that haven't been updated since they've been introduced. I also changed to the old less bulky cleaner computer used before 1.4.1.

developer9998 committed:
Minor fixes & tweaks

developer9998 committed:
Some minor tweaks

- Fixed some Color/Colour tab issues
- Prevented accidental key spam
- Updated RoomView code
- Updated SupportView code

developer9998 committed:
Remove duplicate patch

developer9998 committed:
Update README.md

developer9998 committed:
More tweaks n' such

developer9998 committed:
Merge branch 'main' of https://github.com/developer9998/ComputerInterface

developer9998 committed:
Update ColorSettingView.cs

developer9998 commented:
Bit of an unexpected update, but I got some more work done on it. Note the attached images may not be 100% as they appear ingame. 

<img src="https://user-images.githubusercontent.com/81720436/232622640-b2cb786c-7162-44c5-a77c-343e419afe29.png" width=40% height=40%>
<img src="https://user-images.githubusercontent.com/81720436/232623350-66e32330-db91-4a17-9a2f-febaa18594de.png" width=40% height=40%>
<img src="https://user-images.githubusercontent.com/81720436/232623455-06261f0d-7d7d-4d0f-ad91-8435eb8e4ee3.png" width=40% height=40%>

developer9998 committed:
Update README.md

gizmogoat commented:
I prefer the old monitor
not nostalgia bias btw, I've never used the older versions of CI since I originally used mods on Quest then started playing on PC many many months later once I moved away from integrated graphics

developer9998 committed:
Update README.md

Corrected a ton of words
Heavily inspired by https://github.com/Vadix88/MonkeMapLoader/blob/master/README.md

developer9998 committed:
Fix for new update

- Added computer to Basement
- Fixed keys incorrectly bumping in/out inappropriately
- Made RemoveComputer more stable
- Reordered the main view

developer9998 committed:
Merge branch 'main' of https://github.com/developer9998/ComputerInterface

developer9998 committed:
Update CustomKeyboardKey.cs

developer9998 committed:
Fixed issues with setting name

Usually when joining and then leaving a room

developer9998 committed:
Update CustomComputer.cs

developer9998 commented:
Latest commit fixes computer not spawning in Forest, and places the computer in the new Beach map

beanz00 commented:
i know this isnt the right place to ask but it is the best place i could find, how do i install this if i can? i want to use it but i don't know how to compile it into a mod that works.

developer9998 committed:
Base game data improvement

Noticed that sometimes my colour wouldn't save properly so I've went back and done some small changes (Probably won't help but it's honestly for the best)

developer9998 commented:
Actual question I don't want ignored, is anything going to come out of this or is it just going to be the "Only known method of getting fixed ComputerInterface"

Graicc commented:
I'd like to merge this eventually but there's a lot of things that need to be changed (by me probably) before its in a mergeable state, so its going to take a while to do all of that work

developer9998 commented:
Alright, fair enough 👍

Graicc commented on ComputerInterface/QueueManager.cs:
```diff
@@ -28,19 +25,19 @@ public static IQueueInfo GetQueue()
                 PlayerPrefs.Save();
                 currentQueue = "DEFAULT";
-            }
+            }*/ // most likely the best idea as of right now just in case it's ignored again
```
Was there any bugs this fixed? or just future proofing?

developer9998 replied:
I primarily did it mainly because the last time gamemodes were touched to add Minigames queue this part was forgotten + I'm pretty sure there's an entire interface just for adding custom queues here

Graicc commented on ComputerInterface/Views/GameSettings/CreditsView.cs:
You should be able to delegate all of this to the in game credits view, just set the page size to the right number and query GetPage for whatever page

Graicc committed:
Fix low hanging fruit

developer9998 commented:
> there's a lot of things that need to be changed (by me probably) before its in a mergeable state

What do you mean by this exactly?

Graicc committed:
Prefer enums for error handling

Graicc commented on ComputerInterface/BaseGameInterface.cs:
```diff
@@ -44,51 +42,113 @@ public static Color GetColor()
+                    GetColor(out var r, out var g, out var b);
+                    InitializeNoobMaterial(r, g, b);
+                }
+                catch
+                {
+                    throw new NullReferenceException(); // usually occurs due to the InitializeNoobMaterial method
+                }
```
Since another exception is immediately thrown, does this even do anything?

developer9998 replied:
Ah that, I forgot to remove it when I did fix the method mentioned in the comment

developer9998 replied:
Usually when it would falsely check to see if it was or wasn't null I don't remember

developer9998 commented:
Oh neat

Graicc commented:
> > there's a lot of things that need to be changed (by me probably) before its in a mergeable state
> 
> What do you mean by this exactly?

Things like the above changes, there shouldn't be a singleton for CustomScreenInfo (or really anywhere in CI, that's what Zenject / Dependency Injection is all about), credits view should just ask the base game for the current page, there's some files with only whitespace changes, and presumably other stuff that I don't know about yet

Graicc committed:
Revert whitespace-only changes

Graicc commented on ComputerInterface/Patches/InitializeGamemodePatch.cs:
```diff
@@ -4,6 +4,7 @@
 
 namespace ComputerInterface.Patches
 {
+    // why is this on computer interface? it feels more like a utilla feature
```
Computer interface was doing stuff with modifing gamemodes first
```suggestion
```

developer9998 replied:
Ah makes sense

Graicc committed:
Remove comment

Graicc commented on ComputerInterface/Views/GameSettings/ItemSettingsView.cs:
```diff
@@ -24,54 +23,51 @@ public override void OnShow(object[] args)
-			_selectionHandler.CurrentSelectionIndex = BaseGameInterface.GetItemMode() ? 1 : 0;
-            _insVolumeFloat = BaseGameInterface.GetInstrumentVolume();
-		}
+		void UpdateState() => _insVolumeFloat = BaseGameInterface.GetInstrumentVolume();
```
Getting rid of the first line prevents the selection from being set to the correct value when the view is first shown. It will always show `Enabled`, even if it is `Disabled`.

Graicc committed:
Fix bug with ItemSettingsView initilization

Graicc committed:
Remove room browser

Graicc commented:
Room browser isn't possible anymore because it relies on photon features that no longer exist in Gorilla Tag

developer9998 commented:
Rip room browser

Graicc commented on ComputerInterface/CustomKeyboardKey.cs:
```diff
@@ -74,61 +79,71 @@ public void Init(CustomComputer computer, EKeyboardKey key, Text keyboardText, s
+                var pos = transform.localPosition;
+                pos.y -= KEY_BUMP_AMOUNT;
+                transform.localPosition = pos;
+                collider.center -= new Vector3(0, 0, KEY_BUMP_AMOUNT / 1.125f);
```
Why are you dividing by 1.125? I'm assuming its related to the local scale but I didn't notice any differences after removing it

developer9998 replied:
From adjusting a Unity project of the game, I felt like that change made for easier typing as that from the testing I did essentially made the collider stay in the same location whenever you pressed a key. This change was pre-beachmap update and I know buttons and probably keys were adjusted in some ways (Which includes Forest/Stump computer)

Graicc commented on ComputerInterface/Views/GameSettings/JoinRoomView.cs:
```diff
@@ -16,23 +12,25 @@ namespace ComputerInterface.Views.GameSettings
 
         public override void OnShow(object[] args)
         {
             base.OnShow(args);
+
+            callbacks = new GameObject();
+            callbacks.name = "RoomCallbacks";
+            UnityEngine.Object.DontDestroyOnLoad(callbacks);
+
+            JoinRoomViewCallbacks calllbacksComponent = callbacks.AddComponent<JoinRoomViewCallbacks>();
+            calllbacksComponent.view = this;
+
```
Why did you move this? You shouldn't need to create a new GameObject every time

developer9998 replied:
Not 100% but I think it was causing some issues?

Graicc committed:
Fix more low hanging fruit

Graicc committed:
Clean up computer color settings view

Graicc commented:
The `ComputerSettingsView` can just ask for the `CustomComputer` because Zenject will ensure that it gets it when its constructed by the `ComputerViewFactory`, which avoids the need to have a singleton (this is basically the point of dependency injection)

Graicc committed:
Clean up CreditsView

Graicc committed:
Clean up mic and voice settings

Graicc committed:
Clean up name view

Graicc committed:
Use custom queue as default menu

Graicc committed:
Clean up readme

Graicc committed:
Clean up computer spawning

Graicc commented:
LGTM
